theano/configdefaults.py

```diff
@@ -96,6 +96,29 @@ def __str__(self):
              StrParam(''),
              in_c_key=False)
 
+class ContextsParam(ConfigParam):
+    def __init__(self):
+        def filter(val):
+            if val == '':
+                return val
+            for v in val.split(';'):
+                s = v.split('->')
+                if len(s) != 2:
+                    raise ValueError("Malformed context map: %s" % (v,))
```

Comment:
This is mostly quick validation.  I can add more checks, but I want to let it be flexible in case we want to support things like a 'cpu' context.

I could add the check for multiple context names, but it is simpler to do that where we parse the value.